modules/alia-component/src/qbits/alia/component.clj

```diff
@@ -0,0 +1,131 @@
+(ns qbits.alia.component
+  "This namespace provides a way to "
+  (:require [com.stuartsierra.component :as component]
+            [clojure.spec.alpha         :as s]
+            [qbits.alia                 :as alia]))
+
+(defn illegal!
+  [& msgs]
+  (throw (IllegalArgumentException. ^String (reduce str "" msgs))))
+
+(defn ->cluster
+  "Validate cluster configuration, then build a `Cluster` instance
+   out of it, thanks to alia."
+  [config]
+  (when-not (s/valid? ::alia/cluster-options config)
+    (illegal! (s/explain-str ::alia/cluster-options config)))
```

Comment:
We might just use s/assert here, it would thrown an ex-info with the explain-data